### PR TITLE
CODEOWNERS: Split test/ code owners

### PR DESCRIPTION
**Pull request by pchaigno**
The @cilium/ci team (renamed in this pull request to @cilium/ci-structure) is in charge of the CI infra. (e.g., Jenkinsfile, scripts, etc.), including the Go helper functions and the overall structure of our tests, but it doesn't have the charge of the actual testing logic.

This pull request splits the `test/` code owners across team to better represents the actual owners of the testing logic.

I have been unable to assign the following files to specific teams and they are therefore matched by the @cilium/ci catch-all at the moment:
```
/test/k8sT/Conformance.go
/test/k8sT/istio.go
```

:warning:  @aanm If we agree on the renaming, we will have to change the team's name when we merge this pull request.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -203,19 +203,50 @@ Jenkinsfile.nightly @cilium/ci
...
+/test/k8sT/manifests/externalIPs/ @cilium/loadbalancer @cilium/ci-structure
+/test/k8sT/Services.go @cilium/loadbalancer @cilium/ci-structure
+/test/runtime/lb.go @cilium/loadbalancer @cilium/ci-structure
+# Datapath tests
+/test/bpf/ @cilium/bpf
+/test/k8sT/Bandwidth.go @cilium/bpf @cilium/ci-structure
+/test/k8sT/DatapathConfiguration.go @cilium/bpf @cilium/ci-structure
+/test/k8sT/Verifier.go @cilium/bpf @cilium/ci-structure
+/test/runtime/benchmark.go @cilium/bpf @cilium/ci-structure
+/test/runtime/connectivity.go @cilium/bpf @cilium/ci-structure
```
Comment (pchaigno) [inline CODEOWNERS:226]: For reviewers: I added this file to @cilium/bpf because it's testing the conntrack logic.

File: CODEOWNERS [hunk 2]
```diff
@@ -203,19 +203,50 @@ Jenkinsfile.nightly @cilium/ci
...
+# Policy tests
+/test/k8sT/PoliciesNightly.go @cilium/policy @cilium/ci-structure
+/test/k8sT/Policies.go @cilium/policy @cilium/ci-structure
+/test/k8sT/StressPolicy.go @cilium/policy @cilium/ci-structure
+/test/runtime/Policies.go @cilium/policy @cilium/ci-structure
+# Hubble/monitoring tests
+/test/k8sT/hubble.go @cilium/hubble @cilium/ci-structure
+/test/runtime/monitor.go @cilium/hubble @cilium/ci-structure
```
Comment (pchaigno) [inline CODEOWNERS:234]: Is this okay with @cilium/hubble? This file isn't exactly testing Hubble but I'm expecting changes to that logic are likely to affect Hubble as well... until we deprecate `cilium monitor` :slightly_smiling_face:

File: CODEOWNERS [hunk 3]
```diff
@@ -203,19 +203,50 @@ Jenkinsfile.nightly @cilium/ci
...
+/test/k8sT/KafkaPolicies.go @cilium/proxy @cilium/ci-structure
+/test/runtime/cassandra.go @cilium/proxy @cilium/ci-structure
+/test/runtime/fqdn.go @cilium/proxy @cilium/ci-structure
+/test/runtime/kafka.go @cilium/proxy @cilium/ci-structure
+/test/runtime/memcache.go @cilium/proxy @cilium/ci-structure
+# Misc. tests
+/test/k8sT/Health.go @cilium/health @cilium/ci-structure
+/test/k8sT/cli.go @cilium/cli @cilium/ci-structure
+/test/runtime/kvstore.go @cilium/kvstore @cilium/ci-structure
+/test/k8sT/Identity.go @cilium/agent @cilium/ci-structure
```
Comment (pchaigno) [inline CODEOWNERS:246]: Unsure about that one as well. Is this okay with @cilium/agent?

Comment (aanm) [reply CODEOWNERS:246]: yes

**Review by aanm — APPROVED** [review]

**Review by christarazi — APPROVED** [review]
Thanks, this is a nice change :rocket:.

**Review by joestringer — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
After [discussion with André and Maciej](https://cilium.slack.com/archives/C7PE7V806/p1606990465292400), I've reworked several files in `test/` to enable further codeowner splitting. Please review commit by commit.

I've re-requested all your reviews because you should know better than to review a draft pull request :stuck_out_tongue_closed_eyes:

**Comment by joestringer** [comment]
test-nightly

**Review by qmonnet — APPROVED** [review]

**Comment by pchaigno** [comment]
> test-nightly

After discussion with Maciej, we decided to remove the Nightly test files (except for `PoliciesNightly.go`) because the associated pipelines have been deprecated a while ago. Maciej also suggested removing a couple stale Jenkinsfile; done in the commit before last.

**Comment by pchaigno** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
:rocket:

**Comment by pchaigno** [comment]
retest-4.9

**Review by christarazi — APPROVED** [review]

**Pull request merged**